src/v/cloud_storage/tests/topic_mount_handler_test.cc

```diff
@@ -0,0 +1,254 @@
+#include "cluster/types.h"
+#include "config/configuration.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "test_utils/test.h"
+#include "utils/retry_chain_node.h"
+
+#include <seastar/core/abort_source.hh>
+
+#include <chrono>
+
+using namespace cloud_storage;
+using namespace std::chrono_literals;
+
+namespace {
+static const ss::sstring test_uuid_str = "deadbeef-0000-0000-0000-000000000000";
+static const model::cluster_uuid test_uuid{uuid_t::from_string(test_uuid_str)};
+static const remote_label test_label{test_uuid};
+static const model::topic_namespace test_tp_ns{
+  model::ns{"kafka"}, model::topic{"tp"}};
+static const model::topic_namespace test_tp_ns_override{
+  model::ns{"kafka"}, model::topic{"override"}};
+static ss::abort_source never_abort;
+static constexpr model::cloud_credentials_source config_file{
+  model::cloud_credentials_source::config_file};
+} // namespace
+
+struct TopicMountHandlerFixture
+  : public s3_imposter_fixture
+  , public testing::TestWithParam<bool> {
+    TopicMountHandlerFixture() {
+        pool.start(10, ss::sharded_parameter([this] { return conf; })).get();
+        remote
+          .start(
+            std::ref(pool),
+            ss::sharded_parameter([this] { return conf; }),
+            ss::sharded_parameter([] { return config_file; }))
+          .get();
+    }
+
+    ~TopicMountHandlerFixture() {
+        pool.local().shutdown_connections();
+        remote.stop().get();
+        pool.stop().get();
+    }
+
+    cluster::topic_configuration
+    get_topic_configuration(cluster::topic_properties topic_props) const {
+        auto topic_cfg = cluster::topic_configuration(
+          test_tp_ns.ns, test_tp_ns.tp, 1, 1);
+        topic_cfg.properties = std::move(topic_props);
+        return topic_cfg;
+    }
+
+    ss::sharded<cloud_storage_clients::client_pool> pool;
+    ss::sharded<remote> remote;
+};
+
+TEST_P(TopicMountHandlerFixture, TestMountTopicManifestDoesNotExist) {
+    set_expectations_and_listen({});
+
+    auto topic_props = cluster::topic_properties{};
+    if (GetParam()) {
+        topic_props.remote_topic_namespace_override = test_tp_ns_override;
+    }
+    auto topic_cfg = get_topic_configuration(std::move(topic_props));
+    auto handler = topic_mount_handler(bucket_name, remote.local());
+
+    retry_chain_node rtc(never_abort, 10s, 20ms);
+    auto mount_result = handler.mount_topic(test_label, topic_cfg, rtc).get();
+    ASSERT_EQ(mount_result, topic_mount_result::mount_manifest_does_not_exist);
+}
+
+TEST_P(TopicMountHandlerFixture, TestMountTopicManifestNotDeleted) {
+    set_expectations_and_listen({});
+    retry_chain_node rtc(never_abort, 10s, 20ms);
+
+    const auto path = cloud_storage_clients::object_key{
+      GetParam()
+        ? "migration/recovery/kafka/override/"
+          "deadbeef-0000-0000-0000-000000000000"
+        : "migration/recovery/kafka/tp/deadbeef-0000-0000-0000-000000000000"};
```

Comment:
nit: just noting that you could've used the same path for the override, and used in a different new topic name, in which case you wouldn't have two object keys. I.e. parameterizing the topic name rather than the topic override

Ultimately not important, but maybe it'd be more representative of what a real override would be like.